### PR TITLE
Task/um 720 dev doc for configuring leak sanitizer

### PR DESCRIPTION
**Pull request by kab163**

### CODE REVIEW

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

File: docs/sphinx/developer_guide.rst [hunk 1]
```diff
@@ -28,6 +28,9 @@ Uberenv helps by doing the following:
 
 Uberenv will create a directory ``uberenv_libs`` containing a Spack instance with the required Umpire dependencies installed. It then generates a host-config file (``<config_dependent_name>.cmake``) at the root of Umpire repository.
 
+.. note::
```
Comment (davidbeckingsale) [inline docs/sphinx/developer_guide.rst:31]: 🎉

File: docs/sphinx/developer_guide.rst [hunk 2]
```diff
@@ -77,3 +80,30 @@ It is also possible to use the CI script outside of CI:
...
+
+.. code-block:: bash
+
+  $ srun -ppdebug -N1 --exclusive python scripts/uberenv/uberenv.py --spec="%clang@9.0.0 cxxflags=-fsanitize=address"
+  $ cd build
+  $ cmake -C <path_to>/hc-quartz-toss_3_x86_64_ib-clang@9.0.0.cmake ..
+  $ cmake --build -j
+  $ ASAN_OPTIONS=detect_leaks=1 make test
+  
+This will configure a build with Clang 9.0.0 and the Leak Sanitizer. If there is a leak in one of the tests, it can be useful to gather more information about what happened and more details about where it happened. One way to do this is to run:
```
Comment (davidbeckingsale) [inline docs/sphinx/developer_guide.rst:97]: Could you add a note saying that the CMake file `hc-quartz-....` can be re-used?

**Review by davidbeckingsale — CHANGES_REQUESTED** [review]
Just one quick change. This looks great, thank you!

**Comment by kab163** [comment]
LGTM

**Review by mcfadden8 — APPROVED** [review]
Looks great to me!

**Comment by kab163** [comment]
LGTM

**Review by davidbeckingsale — APPROVED** [review]

**Pull request merged**